### ISSUE TITLE
Error when creating a new table

### ISSUE THREAD

**Issue by creylopez** [nextcloud/tables#847]
### Steps to reproduce

1. Create new table
2. Custom template
3. Assign a name
4. Push Create

### Expected behavior

Table should be created

### Actual behavior

Table is not created and a "Could not insert table. Unknown error" error is raised

### Tables app version

0.6.6

### Browser

Any

### Client operating system

Ubuntu, macOS

### Operating system

Nextcloud official docker image

### Web server

None

### PHP engine version

None

### Database

None

**Comment by juliushaertl** [comment]
Can you check your nextcloud log for an error?

**Comment by creylopez** [comment]
Hi,I tried but couldn’t the logs location. Where can I find them?By the way, if I choose the “task list” template it works. Thanks. Saludos / Best regards / Viele Grüße Celestino ReyEl 15 feb 2024, a las 17:45, Julius Härtl ***@***.***> escribió:﻿
Can you check your nextcloud log for an error?

**Comment by blizzz** [comment]
it is nextcloud.log in the data directory

**Comment by creylopez** [comment]
Ok, thanks, this is what it says when trying to créate a table. As said, any table except the Tasks List template.

{"reqId":"VPYhAdSOb579bfy3VIxL","level":3,"time":"2024-02-15T18:45:37+00:00","remoteAddr":"192.168.1.147","user":"creylopez","app":"tables","method":"POST","url":"/apps/tables/table","message":"An exception occurred while executing a query: SQLSTATE[22007]: Invalid datetime format: 1366 Incorrect string value: '\\xF0\\x9F\\x92\\xBC' for column `nextcloud_db`.`oc_tables_tables`.`emoji` at row 1","userAgent":"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/17.2.1 Safari/605.1.15","version":"28.0.2.5","data":{"app":"tables"}}

Something related to date time format.

I’m Spanish and Nextcloud is configured in Spanish, by the way.

De: Arthur Schiwon ***@***.***>
Fecha: jueves, 15 de febrero de 2024, 19:44
Para: nextcloud/tables ***@***.***>
CC: creylopez ***@***.***>, Author ***@***.***>
Asunto: Re: [nextcloud/tables] Error when creating a new table (Issue #847)

it is nextcloud.log in the data directory

**Comment by blizzz** [comment]
Indeed, but that is strange. The `emoji` column is a string type, not a date time. What database are you using?

**Comment by creylopez** [comment]
I’m using MySQL. Saludos / Best regards / Viele Grüße Celestino ReyEl 15 feb 2024, a las 19:54, Arthur Schiwon ***@***.***> escribió:﻿
Indeed, but that is strange. The emoji column is a string type, not a date time. What database are you using?

**Comment by creylopez** [comment]
By the way, in de configuration section, in general, the following message appears:It says something about MySQL not supporting 4 bytes characters. Could it be that?Saludos / Best regards / Viele Grüße Celestino ReyEl 15 feb 2024, a las 19:54, Arthur Schiwon ***@***.***> escribió:﻿
Indeed, but that is strange. The emoji column is a string type, not a date time. What database are you using?

**Comment by juliushaertl** [comment]
You might want to check https://docs.nextcloud.com/server/latest/admin_manual/configuration_database/mysql_4byte_support.html to enable proper support for 4 byte characters on your setup.

Not entirely sure how we handle this in other apps, either we just skip the emoji or we should block using tables without support for it.

**Comment by creylopez** [comment]
Yes, that was it. I followed the instructions and now I can create tables. Thanks a lot. Saludos / Best regards / Viele Grüße Celestino ReyEl 15 feb 2024, a las 22:13, Julius Härtl ***@***.***> escribió:﻿
You might want to check https://docs.nextcloud.com/server/latest/admin_manual/configuration_database/mysql_4byte_support.html to enable proper support for 4 byte characters on your setup.
Not entirely sure how we handle this in other apps, either we just skip the emoji or we should block using tables without support for it.

**Comment by blizzz** [comment]
It is certainly useful to have a more clear user feedback in that case. In the Admin settings overview, the MySQL setup issue should be shown though. We could re-utilize the result of this check and show a warning on tables, as currently an emoji is stored in any case, and having no emojis/mb4 support in the DB will derogate user experience in Nextcloud in other places as well.

**Comment by enjeck** [comment]
should be fixed
